platforms/software/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -198,48 +182,38 @@ public Property<Boolean> getUseIncubating() {
     @Optional
     @Incubating
     @Option(option = "java-version", description = "Provides java version to use in the project.")
-    public Property<String> getJavaVersion() {
-        return javaVersion;
-    }
+    public abstract Property<String> getJavaVersion();
 
     /**
      * The name of the generated project, defaults to the name of the directory the project is generated in.
      *
-     * This property can be set via command-line option '--project-name'.
-     *
      * @since 5.0
      */
     @Input
-    @ToBeReplacedByLazyProperty
-    public String getProjectName() {
-        return projectName == null ? projectDir.getAsFile().getName() : projectName;
-    }
+    @Optional
+    @Option(option = "project-name", description = "Set the project name.")
+    @ReplacesEagerProperty
+    public abstract Property<String> getProjectName();
```

Comment:
❓ Should we set convention for getProjectName():
```
getProjectName().convention(projectDir.getAsFile().getName())
```
so it's the same as original method?

Reply:
It's a bit complicated. 
Initially, I did a convention, but the current logic relies on the fact that the value was set or not. With current provider API we can't say if the value was changed or not (only if the value is present, which is always true with the conventional value)

Reply:
Oh I see, and projectDir name becomes a default when asking a question... Got it